### ISSUE TITLE
Templater with OCD template giving 'parsing error' 

### ISSUE THREAD

**Issue by Pika78** [SilentVoid13/Templater#711]
Hello, I am getting following the error when trying to OCR a picture  with version 1.12.0 of Templater on Obsidian 0.15.9, using `Templater: Create new note from template` : OCR

`Templater Error: Template parsing error, aborting. tp.user.ocr is not a function`

The template used is the following:

```
---
creation date: <% tp.file.creation_date() %>
tags: [OCR]
---

<%*
const supportedFileTypes = ["jpeg", "jpg", "png"];
const images = this.app.vault.getFiles().filter((item) => supportedFileTypes.indexOf(item.extension) >= 0)
const target = await tp.system.suggester((item) => item.path, images, true);
const out = await tp.user.ocr({ocr_input: target.path});
%><%* tR += out %>
```

Interestingly enough, on the same machine (Macbook Air with mas OS Monterey) with the same version of Obsidian, but in a different Vault where the version of Templater is 1.7.1, the same script as above is working fine.

Is there anything I should do differently or a workaround for this behavior, or do I have to do the OCR in the vault that works and copy the results in the other one?

Many thanks for your help with this issue.
Best regards

**Comment by AB1908** [comment]
Does the script exist in the vault having an error? I feel like you haven't set up your user script correctly. Can you check templater settings to see if the user script is detected or not?

**Comment by Pika78** [comment]
Yes it is in the vault that has the error. I replicated exactly the same structure and content in both vaults. And set the same options for Templater in the vault with the error than in the other.

**Comment by AB1908** [comment]
Can you confirm that the user script shows up in Templater's settings?

**Comment by Pika78** [comment]
In the Templater set up I've got this : 
<img width="897" alt="cap 2022-08-02 à 14 38 50" src="https://user-images.githubusercontent.com/14588340/182376939-29c41fbc-cb74-44dd-a657-52b50f112f7d.png">
<img width="913" alt="cap 2022-08-02 à 14 38 30" src="https://user-images.githubusercontent.com/14588340/182376950-32bf9957-9a12-4609-81d9-b48534c0f62a.png">

and this is where the file with the script rests :
<img width="1216" alt="cap 2022-08-02 à 14 41 27" src="https://user-images.githubusercontent.com/14588340/182377290-c38da909-ed1a-4cc5-a8b3-e7957ec3ec89.png">

Does this answer your question ?

**Comment by AB1908** [comment]
Your error ususally appears when it cannot find the user script, not the user system command. See:
![image](https://user-images.githubusercontent.com/14124383/182378651-1dad97de-7dc3-4aaf-ad4f-e17fb1de0cc3.png)

Also try following these instructions: https://shbgm.ca/obsidian/docs/how-to-use-templater-js-scripts

**Comment by Pika78** [comment]
Many thanks for your help I will look at this resource and get back to you with the results.
